xonsh/completers/bash.py

```diff
@@ -107,7 +111,8 @@ def complete_from_bash(prefix, line, begidx, endidx, ctx):
     script = BASH_COMPLETE_SCRIPT.format(
         filename=fnme, line=' '.join(shlex.quote(p) for p in splt),
         comp_line=shlex.quote(line), n=n, func=func, cmd=cmd,
-        end=endidx + 1, prefix=prefix, prev=shlex.quote(prev))
+        end=endidx + 1, prefix=prefix, prev=shlex.quote(prev),
+        completions_sources='\n'.join(_collect_completions_sources()))
```

Comment:
I think this is doing too much.  We only need the `bash_completions` file that is associated with (in the directory above) the filename we are trying to source. We don't need / want all of the `bash_completions` specified by the user for the whole system.  This could lead to some weird out-of-order errors, in theory.